### PR TITLE
Update to support terraform 0.12

### PR DESCRIPTION
**Pull request by yorinasub17**
__Depends on https://github.com/gruntwork-io/package-terraform-utilities/pull/17__

This is the result of running `terraform 0.12upgrade` and flushing out all the bugs from running the tests locally.

There was one major API difference that was necessary:
For the TLS modules, we no longer cannot pass through the subject info of the TLS cert as an inline block due to type issues. The main issue here is with the `street_address` attribute, which is a list of strings. To support the types, I had to convert this to a string on the input variable, and then reconvert it to a list of strings by using a delimiter. I arbitrarily chose newline for now, but open to other suggestions.

### CODE REVIEW

File: examples/k8s-tiller-kubergrunt-minikube/variables.tf [hunk 1]
```diff
@@ -17,13 +17,12 @@ variable "service_account_name" {
 
 variable "tls_subject" {
   description = "The issuer information that contains the identifying information for the Tiller server. Used to generate the TLS certificate keypairs."
-  type        = "map"
+  type        = map(string)
```
Comment (autero1) [inline examples/k8s-tiller-kubergrunt-minikube/variables.tf]: Wonder if we should use the "typed" variables across the line. Here, for example:
```
type = object({
    common_name    = string
    org = string
  })
```

File: examples/k8s-tiller-kubergrunt-minikube/variables.tf [hunk 2]
```diff
@@ -35,13 +34,12 @@ variable "tls_subject" {
 
 variable "client_tls_subject" {
   description = "The issuer information that contains the identifying information for the helm client of the operator. Used to generate the TLS certificate keypairs."
-  type        = "map"
+  type        = map(string)
```
Comment (autero1) [inline examples/k8s-tiller-kubergrunt-minikube/variables.tf]: Another case of typed variables

File: modules/k8s-helm-client-tls-certs/variables.tf [hunk 3]
```diff
@@ -6,8 +6,9 @@
 # TLS certificate information
 
 variable "tls_subject" {
-  description = "The issuer information that contains the identifying information for the signed certificates. See https://www.terraform.io/docs/providers/tls/r/cert_request.html#common_name for a list of expected keys."
-  type        = "map"
+  description = "The issuer information that contains the identifying information for the signed certificates. See https://www.terraform.io/docs/providers/tls/r/cert_request.html#common_name for a list of expected keys. Note that street_address must be a newline separated string as opposed to a list of strings."
+  # We use an string type here instead of directly specifying the object, to allow certain keys to be optional.
```
Comment (autero1) [inline modules/k8s-helm-client-tls-certs/variables.tf]: So if you specify the object type, all keys are mandatory?

File: examples/k8s-tiller-kubergrunt-minikube/variables.tf (hunk 1 above)
Comment (yorinasub17) [reply examples/k8s-tiller-kubergrunt-minikube/variables.tf]: I was originally thinking about using typed objects, but then after reading the docs further, I found out two limitations:

> Values that match the object type must contain all of the specified keys, and the value for each key must match its specified type.

> Values with additional keys can still match an object type, but the extra attributes are discarded during type conversion.

(source: https://www.terraform.io/docs/configuration/types.html#object-)

So this makes it impossible to specify a type where the keys are optional.

**Comment by yorinasub17** [comment]
Ok added the concrete types and updated to the released version of `package-terraform-utilities`. @autero1 any objections to merging this?

**Comment by autero1** [comment]
No objections, go ahead and merge.

**Comment by yorinasub17** [comment]
UPDATE: Bumped tf version in CI to 0.12.1, and added tfversions badge. Will merge if and when the build passes.

**Comment by yorinasub17** [comment]
Merging!

**Pull request merged**